### ISSUE TITLE
Unused Method Item.getAnimationParameters

### ISSUE THREAD

**Issue by shartte** [MinecraftForge/MinecraftForge#3199]
I've been trying to figure out how to animate an item held by the player (it's the meteorite compass from AE2) and I have come across this method in the Item class added by Forge:

```
public com.google.common.collect.ImmutableMap<String, net.minecraftforge.common.animation.ITimeValue> getAnimationParameters(final ItemStack stack, final World world, final EntityLivingBase entity)
```

The problem is, however, that this method is called nowhere within Forge.
I've tried finding any code on the Internet that uses this method and didn't find any.
Is this an oversight?

**Comment by williewillus** [comment]
It's for the model animation API.

If you want to do compass-like stuff you're better off doing something similar to what the vanilla compass does. See the ItemCompass constructor and the corresponding models/item/compass.json

**Comment by shartte** [comment]
> It's for the model animation API.

But it's not actually used inside of Forge? Is this supposed to be used by mods and not internally by forge, but if so, why? I tried finding any mods on the internet that use it and came up with nothing.

**Comment by williewillus** [comment]
Ask @RainWarrior when he gets back. It's possible the system is not complete.
But as for your original use case, look at how the compass does it.

**Comment by shartte** [comment]
Okay it's fine, but I already found an alternative solution to display my animated 3d item in-hand.

**Comment by Actuarius** [comment]
@williewillus added labels [Cleanup]

**Comment by marvin-roesch** [comment]
I only see this issue now and since I've worked with the animation system a little bit lately, I'd like to explain this despite the issue being closed.

I am fairly certain that this method was added as convenience for *modders* to allow them easy use of property getters as parameters in their `IAnimationStateMachine`.

Hence, I'd say this method can stay anyways (not like it was touched in 1.5 years xD), even if Forge itself doesn't use it and modders could just pass their own time values here:
https://github.com/MinecraftForge/MinecraftForge/blob/4b1b0d83e99ba77f7847fb387017fb744396f8aa/src/main/java/net/minecraftforge/common/model/animation/AnimationStateMachine.java#L216

In summary: Probably not a mistake/oversight by fry, but wise precognition for the one modder that might dare use the system at all ;)
